### PR TITLE
feat(hub-common): enriches channel with groups via hub search includes

### PR DESCRIPTION
**Pull request by tannerjt**
affects: @esri/hub-common

ISSUES CLOSED: #7405

1. Description: Enriches channel result from hubSearch with group information when `include: ['groups']` is requested from `options: IHubSearchOptions`.

1. Instructions for testing: `hubSearchChannels.test.ts`

1. Closes Issues: #7405

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: packages/common/test/search/_internal/hubSearchChannels.test.ts [hunk 1]
```diff
@@ -157,4 +163,93 @@ describe("discussionsSearchItems Module |", () => {
       name: ["Foo"],
     } as any as ISearchChannels);
   });
+  it("exclodes group enrichment when include groups not requested", async () => {
```
Comment (rweber-esri) [inline packages/common/test/search/_internal/hubSearchChannels.test.ts:166]: Minor, typo `excludes`

File: packages/common/test/search/_internal/hubSearchChannels.test.ts [hunk 2]
```diff
@@ -157,4 +163,93 @@ describe("discussionsSearchItems Module |", () => {
...
+      sortField: "createdAt",
+      sortOrder: "desc",
+      requestOptions: {
+        isPortal: false,
+        hubApiUrl: "https://hubqa.arcgis.com/api",
+        token: "my-secret-token",
+      } as IHubRequestOptions,
+      include: ["groups"],
+    };
+    const result = await hubSearchChannels.hubSearchChannels(qry, opts);
```
Comment (rweber-esri) [inline packages/common/test/search/_internal/hubSearchChannels.test.ts:250]: Maybe check that `result` has the value we'd expect

File: packages/common/test/search/_internal/hubSearchChannels.test.ts [hunk 3]
```diff
@@ -157,4 +163,93 @@ describe("discussionsSearchItems Module |", () => {
...
+      num: 10,
+      sortField: "createdAt",
+      sortOrder: "desc",
+      requestOptions: {
+        isPortal: false,
+        hubApiUrl: "https://hubqa.arcgis.com/api",
+        token: "my-secret-token",
+      } as IHubRequestOptions,
+    };
+    await hubSearchChannels.hubSearchChannels(qry, opts);
```
Comment (rweber-esri) [inline packages/common/test/search/_internal/hubSearchChannels.test.ts:191]: Similar, maybe add an assertion that this resolves what we'd expect it to?

File: packages/common/src/search/_internal/hubSearchChannels.ts [hunk 4]
```diff
@@ -137,11 +140,31 @@ export const toHubSearchResult = (
...
+      includes: {
+        // when 'include' requests 'groups' enrichment, get group details
+        groups: options.include?.includes("groups")
+          ? await Promise.all(
+              channel.groups.map(async (groupId) => {
+                let group;
+                try {
+                  group = await getGroup(groupId, options.requestOptions);
+                } catch (e) {
+                  /* tslint:disable-next-line: no-console */
```
Comment (rweber-esri) [inline packages/common/src/search/_internal/hubSearchChannels.ts:152]: Minor, maybe we should resolve `null` vs `undefined`?

**Review by rweber-esri — APPROVED** [review]
LGTM

File: packages/common/test/search/_internal/hubSearchChannels.test.ts (hunk 1 above)
Comment (tannerjt) [reply packages/common/test/search/_internal/hubSearchChannels.test.ts]: Oops, thank you for catching these typos!

**Pull request merged**